kratos/python_scripts/restart_utility.py

```diff
@@ -197,7 +197,7 @@ def _GetRestartFiles(self):
 
             file_name_data_collector = KratosMultiphysics.FileNameDataCollector(self.model_part, os.path.join(self.__GetFolderPathSave(), self._GetFileNamePattern()), {})
             # barrier is necessaryto avoid having some ranks deleting files while other ranks still detect them in the same directory
-            self.model_part.GetCommunicator().Barrier()
+            self.model_part.GetCommunicator().GetDataCommunicator().Barrier()
```

Comment:
@philbucher are you sure this is where the barrier should be?  The filenames are read from the directory when `GetFileNameDataList` is called.

Does this work? My suggestion would be to put the barrier before line 211.

https://github.com/KratosMultiphysics/Kratos/blob/b3d553c77e22d4626dc28a82dcf6f38d3bedff15/kratos/utilities/file_name_data_collector.cpp#L165-L179

Reply:
you are right :)